server/src/domains/auth/auth.service.ts

```diff
@@ -0,0 +1,146 @@
+        const token = await this.tokensRepository.findOneBy({ refreshToken })
+        return token
+    }
+
+    async getTokenByUserId(userId: number): Promise<Token> {
+        const token = await this.tokensRepository.findOneBy({ userId })
+        return token
+    }
+
+    async saveToken(refreshToken: string, userId: number): Promise<Token> {
+        const token = await this.getTokenByUserId(userId)
+        if (token) {
+            token.refreshToken = refreshToken
+            return await this.tokensRepository.save(token)
+        }
+        const user = await this.usersService.getById(userId)
+        const newToken = this.tokensRepository.create({ refreshToken, userId, user })
+        return await this.tokensRepository.save(newToken)
+    }
+
+    async removeTokenByValue(refreshToken: string): Promise<Omit<Token, 'id'>> {
+        try {
+            const token = await this.getTokenByValue(refreshToken)
+            return await this.tokensRepository.remove(token)
+        } catch (e) {
+            return null
+        }
+    }
+
+    async removeTokenByUserId(userId: number): Promise<Omit<Token, 'id'>> {
+        try {
+            const token = await this.getTokenByUserId(userId)
+            return await this.tokensRepository.remove(token)
+        } catch (e) {
+            return null
+        }
+    }
+
+    async signUp(createUserDto: CreateUserDto): Promise<AuthReturnDto> {
+        const candidate = await this.usersService.getByEmail(createUserDto.email)
+        if (candidate) {
+            throw new HttpException(`User with email ${createUserDto.email} is existing`, HttpStatus.BAD_REQUEST)
+        }
+
+        const hashPassword = await bcrypt.hash(createUserDto.password, 3)
+        const user = await this.usersService.create({
+            ...createUserDto,
+            password: hashPassword
+        })
+
+        const userPayload = new UserPayload(user)
+        const tokens = this.generateTokens({ ...userPayload })
+        await this.saveToken(tokens.refreshToken, user.id)
+
+        return { ...tokens, user }
+    }
+
+    async signIn(signInDto: SignInDto): Promise<AuthReturnDto> {
+        const user = await this.usersService.getByEmail(signInDto.email)
+        if (!user) {
+            throw new HttpException('User does not exist', HttpStatus.BAD_REQUEST)
+        }
+
+        const passIsValid = await bcrypt.compare(signInDto.password, user.password)
+
+        if (!passIsValid) {
+            throw new HttpException('Password is wrong', HttpStatus.BAD_REQUEST)
+        }
+
+        const userPayload = new UserPayload(user)
+        const tokens = this.generateTokens({ ...userPayload })
+        await this.saveToken(tokens.refreshToken, user.id)
+
+        return { ...tokens, user }
+    }
+
+    async signOut(refreshToken: string): Promise<Omit<Token, 'id'>> {
+        if (!refreshToken) {
+            throw new UnauthorizedException({ message: 'User is not authorized' })
```

Comment:
Same comment about custom error messages as for server/src/domains/auth/auth.guard.ts file